utilities/workers/worker_manager.py

```diff
@@ -195,3 +213,49 @@ def join_workers(self) -> None:
         """
         for worker in self.__workers:
             worker.join()
+
+    def check_and_restart_dead_workers(self) -> bool:
+        """
+        Check and restart dead workers.
+
+        Returns whether the dead workers were able to be restarted.
+        """
+        new_workers = []
+        for worker in self.__workers:
+            if worker.is_alive():
+                new_workers.append(worker)
+            else:
+                # Log the error
+                frame = inspect.currentframe()
+                target_and_worker_name = (
+                    self.__worker_properties.get_target_name() + " " + worker.name
+                )
+                self.__local_logger.error(
```

Comment:
I think warning or info level is more appropriate, because error level implies almost unrecoverable failure.